src/dex/bebop/bebop.ts

```diff
@@ -0,0 +1,744 @@
+      // "filledTakerAmount":"0"
+      CALLDATA_GAS_COST.AMOUNT
+    );
+  }
+
+  // Encode params required by the exchange adapter
+  // V5: Used for multiSwap, buy & megaSwap
+  // V6: Not used, can be left blank
+  // Hint: abiCoder.encodeParameter() could be useful
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: BebopData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    const { tx } = data;
+
+    if (!tx) {
+      throw new Error('No tx in data');
+    }
+
+    // Encode here the payload for adapter
+    const payload = tx.data;
+
+    return {
+      targetExchange: this.settlementAddress,
+      payload,
+      networkFee: '0',
+    };
+  }
+
+  // This is called once before getTopPoolsForToken is
+  // called for multiple tokens. This can be helpful to
+  // update common state required for calculating
+  // getTopPoolsForToken. It is optional for a DEX
+  // to implement this
+  async updatePoolState(): Promise<void> {
+    const tokens = await this.getCachedTokens();
+
+    if (tokens) {
+      this.tokensMap = tokens;
+    }
+  }
+
+  getMaxLiquidity(levels: BebopLevel[]) {
+    return levels.reduce((acc, [price, size]) => {
+      return acc + price * size;
+    }, 0);
+  }
+
+  // Returns list of top pools based on liquidity. Max
+  // limit number pools should be returned.
+  async getTopPoolsForToken(
+    tokenAddress: Address,
+    limit: number,
+  ): Promise<PoolLiquidity[]> {
+    const prices = await this.getCachedPrices();
+
+    if (!prices) {
+      return [];
+    }
+
+    const pools: PoolLiquidity[] = [];
+
+    for (const [pair, pairData] of Object.entries(prices)) {
+      let liquidityUSD = 0;
+      let token;
+      const [base, quote] = pair.split('/');
+      if (base.toLowerCase() == tokenAddress.toLowerCase()) {
+        const liquidityInQuote = this.getMaxLiquidity(pairData.bids);
+        token = {
+          address: quote,
+          decimals: this.tokensMap[quote.toLowerCase()].decimals,
+        };
+        const quoteTokenUsd = await this.dexHelper.getTokenUSDPrice(
+          token,
+          BigInt(Math.round(liquidityInQuote)),
+        );
+        liquidityUSD = liquidityInQuote * quoteTokenUsd;
+      } else if (quote.toLowerCase() == tokenAddress.toLowerCase()) {
+        const liquidityInBase = this.getMaxLiquidity(pairData.asks);
+        token = {
+          address: base,
+          decimals: this.tokensMap[base.toLowerCase()].decimals,
+        };
+        const baseTokenUsd = await this.dexHelper.getTokenUSDPrice(
+          token,
+          BigInt(Math.round(liquidityInBase)),
+        );
+        liquidityUSD = liquidityInBase * baseTokenUsd;
+      }
+      if (liquidityUSD) {
+        if (pools.length === 0) {
+          pools.push({
+            exchange: this.dexKey,
+            address: this.settlementAddress,
+            connectorTokens: [
+              {
+                address: quote,
```

Comment:
Shouldn't it be `quote` or `base` token based on the direction?

Reply:
updated